source/host_resolver.c

```diff
@@ -1211,6 +1286,10 @@ static void aws_host_resolver_thread(void *arg) {
     aws_array_list_clean_up(&new_address_list);
     aws_array_list_clean_up(&expired_address_list);
 
+    /* trigger the purge complete callback */
+    if (host_entry->on_host_purge_complete != NULL) {
+        host_entry->on_host_purge_complete(host_entry->on_host_purge_complete_user_data);
```

Comment:
Why don't we invoke the callback from `s_on_host_entry_shutdown_completion`? I haven't really went through the code about where to invoke the callback, just asking.

Reply:
The callback is for when host entry is removed from the cache, not when it has shutdown successfully. `aws_thread_current_at_exit` might fail and the callback will never be invoked if we do it in `s_on_host_entry_shutdown_completion`.